library/core/src/intrinsics/mod.rs

```diff
@@ -2763,6 +2763,22 @@ pub const fn size_of<T>() -> usize;
 #[rustc_intrinsic]
 pub const fn align_of<T>() -> usize;
 
+/// The offset of a field inside a type.
+///
+/// Note that, unlike most intrinsics, this is safe to call;
+/// it does not require an `unsafe` block.
+/// Therefore, implementations must not require the user to uphold
+/// any safety invariants.
+///
+/// The stabilized version of this intrinsic is [`core::mem::offset_of`].
```

Comment:
This can also only be called at compile time, right? Please add that note to the comment.

Also, would be good to explain why this is a lang item, since that is very unusual for intrinsics.